client/components/data/query/index.jsx

```diff
@@ -0,0 +1,59 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { PureComponent } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { values } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import compareProps from 'lib/compare-props';
+import { getRequest } from 'state/selectors';
+
+class Query extends PureComponent {
+	static propTypes = {
+		compareOptions: PropTypes.object,
+		requestAction: PropTypes.func.isRequired,
+		requesting: PropTypes.bool.isRequired,
+	};
+
+	componentWillMount() {
+		this.comparator = compareProps( {
```

Comment:
We seem to have some simple query components that wouldn't be using the comparator at all - should we allow for the comparator to be optional then, so it wouldn't be defined if it's not to be used?